apps/publish.py

```diff
@@ -18,6 +19,15 @@
 logger = logging.getLogger(__name__)
 
 
+def get_layers_metadata(factory: str, ver: str, archs: []) -> dict:
+    arch2buildarch = {"amd64": "build-amd64", "arm64": "build-aarch64", "arm": "build-armhf"}
+    layers_meta = {}
+    project = f"{factory}/lmp" if factory != "lmp" else "lmp"
+    for arch in archs:
+        layers_meta[arch] = jobserv_get(f"/projects/{project}/builds/{ver}/runs/{arch2buildarch[arch]}/layers_meta.json")
```

Comment:
is this safe for a factory that doesn't build for all architectures?

Reply:
I guess no. I've forgotten about `SKIP_ARCHS`.

Reply:
Yes, it is. The list of the build runs corresponds to the `MANIFEST_PLATFORMS_DEFAULT` env. var. and to the `archs` list passed as an input param to this function https://github.com/foundriesio/ci-scripts/pull/285/files#diff-008d54a346d8cbbcee50d5e9a66d7e32feba786085ab3248822aea5d8070705cR22

Reply:
`SKIP_ARCHS` is already taken into account by the current code in the compose publish that gather App layers and adds layer metadata in a form of index manifest. Effectively, the code just don't add layers meta for the arch for which at least one of the App images was "skipped".

Reply:
i didn't believe you and re-read all the code and ... shocker - you are right :)